src/main/sensors/barometer.c

```diff
@@ -475,40 +440,33 @@ uint32_t baroUpdate(timeUs_t currentTimeUs)
     return sleepTime;
 }
 
-static float pressureToAltitude(const float pressure)
-{
-    return (1.0f - powf(pressure / 101325.0f, 0.190295f)) * 4433000.0f;
-}
-
-int32_t baroCalculateAltitude(void)
+float baroCalculateAltitude(void)
 {
-    int32_t BaroAlt_tmp;
-
     // calculates height from ground via baro readings
     if (baroIsCalibrationComplete()) {
-        BaroAlt_tmp = lrintf(pressureToAltitude((float)(baroPressureSum / barometerConfig()->baro_sample_count)));
-        BaroAlt_tmp -= baroGroundAltitude;
-        baro.BaroAlt = lrintf((float)baro.BaroAlt * CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_noise_lpf) + (float)BaroAlt_tmp * (1.0f - CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_noise_lpf))); // additional LPF to reduce baro noise
-    }
-    else {
-        baro.BaroAlt = 0;
+        baro.BaroAlt = filteredBaroAltitudeRaw - baroGroundAltitude;
+    } else {
+        baro.BaroAlt = 0.0f;
     }
+    DEBUG_SET(DEBUG_BARO, 3, baro.BaroAlt); // metres * 100
     return baro.BaroAlt;
 }
 
 void performBaroCalibrationCycle(void)
 {
-    static int32_t savedGroundPressure = 0;
-
-    baroGroundPressure -= baroGroundPressure / 8;
-    baroGroundPressure += baroPressureSum / barometerConfig()->baro_sample_count;
-    baroGroundAltitude = (1.0f - pow_approx((baroGroundPressure / 8) / 101325.0f, 0.190259f)) * 4433000.0f;
-
-    if (baroGroundPressure == savedGroundPressure) {
+    static float savedGroundAltitude = 10000000.0f; // some non-zero value
```

Comment:
This was a hack because if `savedGroundAltitude` happened randomly to be a value that caused `fabsf(baroGroundAltitude - savedGroundAltitude) < 0.1f)` to be true, then that initial value would be accepted as the cal value before the smoothed value was set.
I was not sure how stop this, so I initialised `savedGroundAltitude` to an impossible value, to ensure failure on the first iteration.
It needs at least one iteration for the smoothed value to be calculated the incoming data.
Maybe we should could count a certain number of iterations before commencing the comparison?
What would you suggest?

Reply:
Maybe use `calibratingB` threshold to force some minimum calibration time